query/btccheckpoint.go

```diff
@@ -1,58 +1,70 @@
 package query
 
 import (
+	"context"
+
 	btcctypes "github.com/babylonchain/babylon/x/btccheckpoint/types"
-	"github.com/babylonchain/rpc-client/client"
+	"github.com/cosmos/cosmos-sdk/client"
 	sdkquerytypes "github.com/cosmos/cosmos-sdk/types/query"
 )
 
-// BTCCheckpointParams queries btccheckpoint module's parameters via ChainClient
-func BTCCheckpointParams(c *client.Client) (*btcctypes.Params, error) {
-	ctx, cancel := c.GetDefaultQueryContext()
+func (c *QueryClient) QueryBTCCheckpoint(f func(ctx context.Context, queryClient btcctypes.QueryClient)) {
```

Comment:
Do we want to make it private? Also, maybe a docstring would be helpful.

This comment also applies to other modules

Reply:
Yep added docstrings for these functions. Maybe keeping them public allows better extensibility. For example, other projects can use this function to define its own queries (e.g., a certain combination of existing APIs) by using this function, without the need of forking/modifying `rpc-client`.